File: src/main/java/calebabg/abstractions/LocalAudioSource.java
package calebabg.abstractions;

public class LocalAudioSource implements IAudioResource {
    private final String title;
    private final String resourcePath;

    public LocalAudioSource(String title, String resourcePath) {
        this.title = title;
        this.resourcePath = resourcePath;
    }

    public String id() { return ""; }

    public String title() {
        return title;
    }

    public String mediaPath() {
        return resourcePath;
    }

    public AudioResourceType resourceType() {
        return AudioResourceType.LOCAL_RESOURCE;
    }
}

File: src/main/kotlin/com/calebabg/abstractions/LocalAudioSource.kt
package com.calebabg.abstractions

class LocalAudioSource(private val title: String, private val resourcePath: String) : AudioResource {
    override fun id(): String = ""
    override fun title(): String = title
    override fun mediaPath(): String = resourcePath
    override fun resourceType(): AudioResourceType = AudioResourceType.LOCAL_RESOURCE
}